src/Time.svelte

```diff
@@ -1,67 +1,69 @@
 <script>
   // @ts-check
 
-  /**
-   * Original timestamp
-   * @type {import("dayjs").ConfigType}
-   */
-  export let timestamp = new Date().toISOString();
+  const {
+    /**
+     * Original timestamp
+     * @type {import("dayjs").ConfigType}
+     */
+    timestamp = new Date().toISOString(),
 
-  /**
-   * Timestamp format for display.
-   * It's also used as a title in the `relative` mode
-   * @type {import("dayjs").OptionType}
-   * @example "YYYY-MM-DD"
-   */
-  export let format = "MMM DD, YYYY";
+    /**
+     * Timestamp format for display.
+     * It's also used as a title in the `relative` mode
+     * @type {import("dayjs").OptionType}
+     * @example "YYYY-MM-DD"
+     */
+    format = "MMM DD, YYYY",
 
-  /**
-   * Set to `true` to display the relative time from the provided `timestamp`.
-   * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
-   * @type {boolean}
-   */
-  export let relative = false;
+    /**
+     * Set to `true` to display the relative time from the provided `timestamp`.
+     * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
+     * @type {boolean}
+     */
+    relative = false,
 
-  /**
-   * Set to `true` to update the relative time at 60 second interval.
-   * Pass in a number (ms) to specify the interval length
-   * @type {boolean | number}
-   */
-  export let live = false;
-
-  /**
-   * Formatted timestamp.
-   * Result of invoking `dayjs().format()`
-   * @type {string}
-   */
-  export let formatted = "";
+    /**
+     * Set to `true` to update the relative time at 60 second interval.
+     * Pass in a number (ms) to specify the interval length
+     * @type {boolean | number}
+     */
+    live = false,
+    ...rest
+  } = $props();
 
   import { dayjs } from "./dayjs";
-  import { onMount } from "svelte";
-
-  /** @type {undefined | NodeJS.Timeout} */
-  let interval = undefined;
 
   const DEFAULT_INTERVAL = 60 * 1_000;
 
-  onMount(() => {
+  $effect(() => {
+    /** @type {undefined | NodeJS.Timeout} */
+    let interval;
+    if (relative && live !== false) {
+      interval = setInterval(
+        () => {
+          formatted = dayjs(timestamp).from();
+        },
+        Math.abs(typeof live === "number" ? live : DEFAULT_INTERVAL),
+      );
+    }
     return () => clearInterval(interval);
   });
```

Comment:
I just caught that

Reply:
good call. it does work, but i agree, it doesn't makes sense for me to update a derived value. i'm thinking `formatted` should be a `$state` instead of a `$derived`. if that makes sense, let me know and i will update.

but, in this instance, the derived will only be updated if `relative`, `live`, or `timestamp` ever change, which, most likely, they won't change.

Reply:
Either is fine, I think that if we update it the initial state needs to be set in the effect?

Reply:
i found that in `$effect` any values that are read asynchronously, for example in a `setInterval`, will not be tracked.

https://svelte.dev/docs/svelte/$effect#Understanding-dependencies

Reply:
also, it doesn't look like it is recommended to set the initial state inside of the `$effect`. my intellij complains with the following:
![image](https://github.com/user-attachments/assets/de2e47c9-5c4b-4e96-b52a-6937e8dd1804)

Reply:
Probably using state would be good, whatever keeps it simple and easy to understand

Reply:
done